src/main/java/com/google/devtools/build/lib/remote/RemoteSpawnRunner.java

```diff
@@ -320,19 +345,56 @@ public SpawnResult exec(Spawn spawn, SpawnExecutionContext context)
     }
   }
 
+  /** conversion utility for protobuf Timestamp difference to java.time.Duration */
+  private static Duration between(Timestamp from, Timestamp to) {
+    return Duration.ofNanos(Durations.toNanos(Timestamps.between(from, to)));
+  }
+
+  @VisibleForTesting
+  static void spawnMetricsAccounting(SpawnMetrics.Builder spawnMetrics, ExecutedActionMetadata executionMetadata) {
+    if (!executionMetadata.getWorker().isEmpty()) {
+      // Accumulate queueTime from any previous attempts
```

Comment:
Can you add a comment on what happens when some of these aren't set?

Reply:
Commenting here on network time, UI won't let me address that individually.

I'm not sure why "Good Enough" is good enough here, especially when it comes to blaming the network for things that bazel is doing in between. I see the logic - technically we can't know how much lag occurs throughout the action - but with each concurrent request happening and being metered separately for uploads and downloads' times, this is a solid tool for measuring the explicit wait time for responses after the requests have gone out.

Misleading cost measurement tools cost more than not having them at all, and I want to avoid blaming the network if it's full GC occurring attributable to Other.

Reply:
You can't control which bucket a full GC gets attributed to. If Bazel starts a full GC after the reply came back, but before you looked at the clock, then it gets officially counted as network time. There's no way for us to distinguish between the cases where the full GC delayed processing (and shouldn't be counted as network time), and where it happened while a request was in-flight, but the GC didn't actually delay processing.

I'm just concerned about the amount of complexity the network timing introduces, and it's not clear to me that it's worth it. At the same time, it's important to get this change in, because we currently have *no*  visibility into the metrics, and wall-time is just wrong.